plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/handlers/UntypeSubApplicationHandler.java

```diff
@@ -55,22 +56,25 @@ public void setEnabled(final Object evaluationContext) {
 	}
 
 	private static SubApp getSelectedSubApp(final Object selection) {
-		if (selection instanceof final IStructuredSelection structSel) {
-			if (!structSel.isEmpty() && (structSel.size() == 1)) {
-				return getSubApp(structSel.getFirstElement());
-			}
+		if (selection instanceof TreeSelection) {
+			return null;
+		}
+		if ((selection instanceof final IStructuredSelection structSel)
+				&& (!structSel.isEmpty() && (structSel.size() == 1))) {
+			return getSubApp(structSel.getFirstElement());
 		}
 		return null;
 	}
 
 	private static SubApp getSubApp(final Object currentElement) {
-		if (currentElement instanceof SubApp) {
-			return (SubApp) currentElement;
+		if (currentElement instanceof final SubApp subApp) {
+			return subApp;
+		}
```

Comment:
should we remove the subapp check? because currently an editor selection always involves an editpart